receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -286,22 +386,61 @@ buffer_count: 2
 		},
 	}
 
-	for _, md := range gotMDs {
-		node := md.Node
-		if diff := cmpNodePb(node, wantNode); diff != "" {
-			t.Errorf("Mismatch Node\n-Got +Want:\n%s", diff)
-		}
-		metricPbs := md.Metrics
-		if len(metricPbs) != 1 {
-			t.Errorf("Want 1 metric, got %d", len(metricPbs))
+	// Firstly sort them so that comparisons return stable results.
+	byMetricsSorter(t, got)
+	byMetricsSorter(t, want1)
+	byMetricsSorter(t, want2)
+
+	if reflect.DeepEqual(got, want1) {
+		fmt.Println("done")
+		return
+	}
+
+	// Since these tests rely on underdeterministic behavior and timing that's imprecise.
+	// The best that we can do is provide any of variants of what we want.
+	wantPermutations := [][]data.MetricsData{
+		want1, want2,
+	}
+
+	for _, want := range wantPermutations {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("different metric got:\n%v\nwant:\n%v\n", string(exportertest.ToJSON(got)),
+				string(exportertest.ToJSON(want)))
```

Comment:
is the purpose here to fail if either want1 or want2 doesn't match? (assuming that want1 and want2 are different. see prev comment. they don't seem to be different)

Reply:
This test was inherited from the original implementation of pmreceiver,  @odeke-em  might have more insight on this test.